src/generator/protocol/operations.ts

```diff
@@ -307,16 +314,17 @@ function createProtocolRequest(client: string, op: Operation, imports: ImportMan
   let text = `${comment(name, '// ')} creates the ${info.name} request.\n`;
   text += `func (${client}) ${name}(${generateParamsSig(sig.protocolSigs.requestMethod.params, true)}) (${sig.protocolSigs.requestMethod.returns.join(', ')}) {\n`;
   text += `\turlPath := "${op.requests![0].protocol.http!.path}"\n`;
-  if (values(aggregateParameters(op)).any((each: Parameter) => { return each.protocol.http!.in === 'path' })) {
-    // replace path parameters
+  const inPathParams = values(aggregateParameters(op)).where((each: Parameter) => { return each.protocol.http !== undefined; }).where((each: Parameter) => { return each.protocol.http!.in === 'path'; });
+  // replace path parameters
+  for (const pp of values(inPathParams)) {
+    // TODO this needs to be fixed, this is an ugly solution to guarantee that string and url imports are only added when an in path parameter is actually added
```

Comment:
I assume this TODO is about repeated calls to `imports.add()`?  You could move this out of the `for` loop, under something like `if (inPathParams.length >0) then add imports`.

Reply:
Yes it's due to the repeated imports and I previously had exactly what you're suggesting but there were failures and edge cases that were causing errors. So I'm digging into this as part of the pending paths fix.

Reply:
For instance in some cases inPathParams.length >0 BUT when we want to iterate over the inPathParams values there's nothing to iterate over, therefore the imports are added but unused.

Reply:
This is done for swaggers that use http.in = path. This does not consider x-ms-paths yet. But did leave the imports outside of the loop. There's an issue here #296  tracking the x-ms-paths functionality.